src/main/java/data/plans/PlanList.java

```diff
@@ -0,0 +1,590 @@
+            planObject = getPlanFromIndexNum(i);
+            isWorkoutInPlan = checkWorkoutInPlan(oldWorkoutToCheck, updatedWorkoutToCheck, planObject);
+            if (isWorkoutInPlan) {
+                planIndexWithTargetWorkout.add(i);
+            }
+        }
+        return planIndexWithTargetWorkout;
+    }
+
+    /**
+     * Gets the number of plans which includes the deleted workout.
+     *
+     * @param workoutToCheck The workout to look for in the plan.
+     * @return An Arraylist of integer which includes the number of the
+     *         plan that includes the deleted workout.
+     * @throws ArrayIndexOutOfBoundsException For operations which involves index checking.
+     */
+    public ArrayList<Integer> findPlanContainsTargetWorkout(String workoutToCheck) throws
+            ArrayIndexOutOfBoundsException {
+        Plan planObject;
+        boolean isWorkoutInPlan = false;
+        ArrayList<Integer> planIndexWithTargetWorkout = new ArrayList<Integer>();
+        for (int i = 1; i <= getPlansDisplayList().size(); i++) {
+            planObject = getPlanFromIndexNum(i);
+            isWorkoutInPlan = checkWorkoutInPlan(workoutToCheck, planObject);
+            if (isWorkoutInPlan) {
+                planIndexWithTargetWorkout.add(i);
+            }
+        }
+        return planIndexWithTargetWorkout;
+    }
+
+    /**
+     * Deletes the plans which contains the deleted workouts.
+     *
+     * @param deletedWorkoutDetail The deleted workout.
+     * @throws ArrayIndexOutOfBoundsException If index .
+     * @throws InvalidPlanException For operations which involves index checking.
+     */
+    public void deletePlanContainsDeletedWorkout(String deletedWorkoutDetail) throws InvalidPlanException {
+        ArrayList<Integer> planIndexWithDeletedWorkout = findPlanContainsTargetWorkout(deletedWorkoutDetail);
+        if (planIndexWithDeletedWorkout.size() <= 0) {
+            return;
+        }
+        if (planIndexWithDeletedWorkout.size() > 0) {
+            System.out.println(deletedWorkoutDetail + " is found in:\n");
+        }
+
+        for (int planNumber : planIndexWithDeletedWorkout) {
+            System.out.println("\t" + getPlansDisplayList().get(planNumber - 1));
+            assert (checkPlanIndexIsWithinRange(planNumber)) : "Plan number is out of range.";
+        }
+        int totalNumberOfPlanToDelete = planIndexWithDeletedWorkout.size();
+        for (int i = 0; i < totalNumberOfPlanToDelete; i++) {
+            if (i == 0) {
+                System.out.println("\nThe following plan has been removed:\n");
+            }
+            System.out.println((i + 1) + ". " + getPlansDisplayList().get(planIndexWithDeletedWorkout.get(i) - i - 1));
+            deletePlan(Integer.toString(planIndexWithDeletedWorkout.get(i) - i));
+        }
+        System.out.println();
+    }
+
+    public void updatePlanContainsUpdatedWorkout(String oldWorkoutToCheck, Workout updatedWorkout) throws
```

Comment:
errr... bad name for a method. 
if I have to trust the name, you are doing 2 things (or more) in this method. refactor appropriately